### PR TITLE
Refactor OL operator

### LINKED ISSUE #119: Refactor to use the common library from Runtime Component Operator
Switch from using Appsody Operator as the base to using Runtime Component Operator v0.4.0.

### PR DESCRIPTION
**Pull request by edavidj**
**What this PR does / why we need it?**:

- Point to generic operator common libraries
- Add images, applications, cert-manager, and service binding changes to OL operator
- Update role files

**Does this PR introduce a user-facing change?**

- [ ] User guide
- [ ] `CHANGELOG.md`

**Which issue(s) this PR fixes**:

Fixes #119

### CODE REVIEW

**Comment by leochr** [comment]
@navidsh @arturdzm Please review when you get a moment. Thanks.

**Review by arturdzm — COMMENTED** [review]
```
	predSubResource := predicate.Funcs{
		UpdateFunc: func(e event.UpdateEvent) bool {
			// Ignore updates to CR status in which case metadata.Generation does not change
			return e.MetaOld.GetGeneration() != e.MetaNew.GetGeneration() && (isClusterWide || watchNamespacesMap[e.MetaOld.GetNamespace()])
		}
```
This needs to be updated, generation check might prevent certificate status notifications to be delievered

File: deploy/role.yaml [hunk 1]
```diff
@@ -84,3 +73,21 @@ rules:
   - services
   verbs:
   - '*'
+- apiGroups:
```
Comment (arturdzm) [inline deploy/role.yaml]: - routes/custom-host is missing in this file

File: pkg/apis/openliberty/v1beta1/openlibertyapplication_types.go [hunk 2]
```diff
@@ -42,10 +43,15 @@ type OpenLibertyApplicationSpec struct {
 	CreateKnativeService *bool                             `json:"createKnativeService,omitempty"`
 	Monitoring           *OpenLibertyApplicationMonitoring `json:"monitoring,omitempty"`
 	CreateAppDefinition  *bool                             `json:"createAppDefinition,omitempty"`
+	ApplicationName      string                            `json:"applicationName,omitempty"`
 	// +listType=map
 	// +listMapKey=name
-	InitContainers []corev1.Container                    `json:"initContainers,omitempty"`
-	Serviceability *OpenLibertyApplicationServiceability `json:"serviceability,omitempty"`
+	InitContainers []corev1.Container `json:"initContainers,omitempty"`
```
Comment (navidsh) [inline pkg/apis/openliberty/v1beta1/openlibertyapplication_types.go]: (not directly related to this line)
@arturdzm do you know why the following is different from Runtime Component operator:
```
	// +listType=map		// +listType=map
	// +listMapKey=name		// +listMapKey=name
	VolumeMounts        []corev1.VolumeMount          `json:"volumeMounts,omitempty"`
```
vs
```
	// +listType=atomic
	VolumeMounts        []corev1.VolumeMount         `json:"volumeMounts,omitempty"`
```

File: pkg/apis/openliberty/v1beta1/openlibertyapplication_types.go [hunk 3]
```diff
@@ -65,12 +71,16 @@ type OpenLibertyApplicationService struct {
...
+	PortName   string `json:"portName,omitempty"`
+	TargetPort *int32 `json:"targetPort,omitempty"`
 
 	Annotations map[string]string `json:"annotations,omitempty"`
 	// +listType=atomic
-	Consumes []ServiceBindingConsumes `json:"consumes,omitempty"`
-	Provides *ServiceBindingProvides  `json:"provides,omitempty"`
+	Consumes             []ServiceBindingConsumes `json:"consumes,omitempty"`
+	Provides             *ServiceBindingProvides  `json:"provides,omitempty"`
+	Certificate          *Certificate             `json:"certificate,omitempty"`
```
Comment (navidsh) [inline pkg/apis/openliberty/v1beta1/openlibertyapplication_types.go]: Missing `// +k8s:openapi-gen=true`

File: pkg/apis/openliberty/v1beta1/openlibertyapplication_types.go [hunk 4]
```diff
@@ -597,6 +688,47 @@ func (s *OpenLibertyApplicationStatus) SetCondition(c common.StatusCondition) {
 	}
 }
 
+// GetAnnotations returns route annotations
+func (r *OpenLibertyApplicationRoute) GetAnnotations() map[string]string {
```
Comment (navidsh) [inline pkg/apis/openliberty/v1beta1/openlibertyapplication_types.go]: Please ensure `SetCondition` code matches RC operator

File: pkg/apis/openliberty/v1beta1/openlibertyapplication_types.go [hunk 5]
```diff
@@ -474,6 +560,7 @@ func (cr *OpenLibertyApplication) GetLabels() map[string]string {
 		"app.kubernetes.io/instance":   cr.Name,
 		"app.kubernetes.io/name":       cr.Name,
 		"app.kubernetes.io/managed-by": "open-liberty-operator",
+		"app.kubernetes.io/component":  "backend",
```
Comment (navidsh) [inline pkg/apis/openliberty/v1beta1/openlibertyapplication_types.go]: This is missing `part-of` and service binding label from before

**Comment by leochr** [comment]
@edavidj Thanks for the PR. 

- In Role and ClusterRole, is `pod/exec` needed for apiGroup `openliberty.io`? 
- test/e2e/openliberty_trace.go - Size changed from 1Gi to 1Mi - any particular reason? is 1Mi enough?

**Comment by edavidj** [comment]
@leochr good catch, I had changed it for a local test with my own volume claim that was a smaller size. I've reverted that and gotten rid of the `pod/exec` under `openliberty.io` in the cluster role.

**Review by navidsh — APPROVED** [review]

**Review by leochr — APPROVED** [review]

**Pull request merged**